pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -84,9 +129,48 @@ func (r *CloudEntitySelectorReconciler) SetupWithManager(mgr ctrl.Manager) error
 		}); err != nil {
 		return err
 	}
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&cloudv1alpha1.CloudEntitySelector{}).
-		Complete(r)
+
+	if err := ctrl.NewControllerManagedBy(mgr).For(&cloudv1alpha1.CloudEntitySelector{}).Complete(r); err != nil {
+		return err
+	}
+	return mgr.Add(r)
+}
+
+func (r *CloudEntitySelectorReconciler) Start(context.Context) error {
+	if r.initialized {
+		r.Log.Info(controllerDuplicateInit)
+		return nil
+	}
+	// Is cache synced?
+	if !(*r.Mgr).GetCache().WaitForCacheSync(context.TODO()) {
+		return fmt.Errorf("failed to sync shared informer cache")
+	}
+
+	for {
+		// Blocking call to wait until CPAController is synced.
+		r.Log.Info(controllerSyncDependentWait, "controller", CPAController)
+		if err := waitForControllersToSync([]string{CPAController}, syncTimeout); err != nil {
+			r.Log.Info("Retry after 1s", "err", err)
+			time.Sleep(1 * time.Second)
+			continue
+		}
+		r.Log.Info(controllerSyncDependentDone, "controller", CPAController)
+		break
+	}
+
+	if err := r.setPendingSync(); err != nil {
+		// retry here? - TODO check if we return error what happens?
+		r.Log.Info("failed to set pending sync count, err %v", err)
+		return err
+	}
+	if r.pendingSyncCount == 0 {
+		r.Log.Info(controllerNothingToSync)
```

Comment:
controllerNothingToSync, too heavily used.. have description rather than constant.. If its used in unit test, pattern match should be Ok..

Reply:
Modified the implementation, this no longer applies..